server/src/main/java/org/opensearch/extensions/ExtensionsOrchestrator.java

```diff
@@ -267,6 +287,105 @@ TransportResponse handleExtensionRequest(ExtensionRequest extensionRequest) thro
         throw new Exception("Handler not present for the provided request");
     }
 
+    /**
+     * Iterates through all discovered extensions, sends transport requests for named writeables and consolidates all entires into a central named writeable registry for extensions.
+     */
+    public void getNamedWriteables() {
+
+        // Retrieve named writeable registry entries from each extension
+        for (DiscoveryNode extensionNode : extensionsInitializedList) {
+            try {
+                Map<DiscoveryNode, Map<Class, Map<String, ExtensionReader>>> extensionRegistry = getNamedWriteables(extensionNode);
+                if (extensionRegistry.isEmpty() == false) {
+                    this.extensionNamedWriteableRegistry.putAll(extensionRegistry);
+                }
+            } catch (UnknownHostException e) {
+                logger.error(e.toString());
+            }
+        }
+
+        // TODO : Invoke during the consolidation of named writeables within Node.java and return extension entries these
+        // (https://github.com/opensearch-project/OpenSearch/issues/4067)
+    }
+
+    /**
+     * Sends a transport request for named writeables to an extension, identified by the given DiscoveryNode, and processes the response into registry entries
+     *
+     * @param extensionNode DiscoveryNode identifying the extension
+     * @throws UnknownHostException if connection to the extension node failed
+     * @return A map of category classes and their associated names and readers for this discovery node
+     */
+    public Map<DiscoveryNode, Map<Class, Map<String, ExtensionReader>>> getNamedWriteables(DiscoveryNode extensionNode)
```

Comment:
Does this method have to be public?

Reply:
Currently I target this method within my unit tests since `extensionsInitializedList` cannot be populated unless extensionsInitialize() occurs. The method which calls this iterates through initialized discovery nodes and sends a request to the extension associated with the node, therefore, in order to test this, I would have to call this method directly, which is only possible if it is public